pkg/reconciler/namespace/namespace.go

```diff
@@ -262,3 +268,41 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, ns *corev1.Namespace)
 	// Don't update anything that is already present.
 	return current, nil
 }
+
+// cleanup of unwanted brokers, service account and rolebinding in namespace with no injection label.
+func (r *Reconciler) cleanupResources(ctx context.Context, ns string) error {
+	// delete broker
+	err := r.EventingClientSet.EventingV1alpha1().Brokers(ns).DeleteCollection(
+		&metav1.DeleteOptions{},
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", resources.ResourceInjectionLabel, "true"),
+			FieldSelector: fmt.Sprintf("metadata.name=%s", resources.DefaultBrokerName),
+		})
+	if err != nil && !k8serrors.IsNotFound(err) {
+		logging.FromContext(ctx).Error("Error deleting broker in namespace", zap.String("brokername", resources.DefaultBrokerName), zap.Error(err))
+		return err
+	}
+
+	serr := r.KubeClientSet.CoreV1().ServiceAccounts(ns).DeleteCollection(
+		&metav1.DeleteOptions{},
+		metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", resources.ResourceInjectionLabel, "true"),
+			FieldSelector: fmt.Sprintf("metadata.name=%s", resources.ServiceAccountName),
+		})
+	if serr != nil && !k8serrors.IsNotFound(serr) {
+		logging.FromContext(ctx).Error("Error deleting service account %s in namespace", zap.String("name", resources.ServiceAccountName), zap.Error(serr))
```

Comment:
And here? perhaps not an error. Log it and try to delete the other things too. 

If one resource is missing you will miss out of deleting the others.

Reply:
Thats a good idea. I will update that